wifiphisher/pywifiphisher.py

```diff
@@ -667,7 +667,9 @@ def start(self):
             # make sure interfaces are not blocked
             self.network_manager.unblock_interface(ap_iface)
             self.network_manager.unblock_interface(mon_iface)
-            self.network_manager.set_interface_mode(mon_iface, "monitor")
+            # set monitor mode only when --essid is not given
+            if self.advanced_enabled() or args.essid is None:
```

Comment:
I agree we should not set the card to monitor mode when the i) ESSID is explicitly set and ii) we only have one card available. That means `args.essid is not None and not self.advanced_enabled()`, right?

Reply:
yes. So the counterpart of this equation `arg.essid is not None and not self.advance_enabled()` means to `set monitor mode`.

The equation for setting monitor mode:
```
NOT(args.essid is not None and not self.advanced_enabled())
<-> NOT(args.essid is not None) or NOT(not self.advanced_enabled())
<-> args.essid is None or self.advanced_enabled()
```
Which is the equation I write in the code.

Under the case `wifiphisher -nJ` we still need to set the card to monitor mode if we only have one card. Since we still need to do the ap-selection.